plugin/hosts/hostsfile.go

```diff
@@ -159,9 +161,9 @@ func ipVersion(s string) int {
 
 // LookupStaticHostV4 looks up the IPv4 addresses for the given host from the hosts file.
 func (h *Hostsfile) LookupStaticHostV4(host string) []net.IP {
-	h.Lock()
-	defer h.Unlock()
 	h.ReadHosts()
```

Comment:
`h.ReadHosts()` can execute without a lock?

Reply:
`h.Readhosts()` only checks to see if the file has been modified, if it has it calls `h.Parse()`. Which is the only function that modifies the maps. I added the `h.Lock()` to that function.

Reply:
Hmm.. looking at it closer, it does get the len() of one of the maps. I was hoping to limit the write lock to only writing the maps, but maybe I should take it in at the beginning of `h.ReadHosts()`.

Reply:
Ah, I pressed approve too soon :) Yes, len need RLock

Reply:
Sorry about that. I am fixing that and making it cleaner.